### PR TITLE
Improvement/vltclt 33 typescript bumping

### PR DESCRIPTION
**Pull request by KazToozs**
- Bumps arsenal with most recent httpagent
- add typsecript transpilation of IAMClient, with small changes
- bump package version
EDIT: add a fix for accountName as expected is JSON from Vault2

Build is green in Vault2 https://github.com/scality/vault2/pull/117

### CODE REVIEW

**Review by anurag4DSB — CHANGES_REQUESTED** [review]
Can we have an update in vault2 with a green CI before merging?
You can use the commit hash(HEAD) of this branch, update in vault and check if the Vault2 CI is green.

Please do add the results in the description of the PR.

Edit:
1. Checkout a branch (with ticket) for Vault2
2. update this vaultclient package with the HEAD of this branch
3. check if the CI is green
4. If the CI is green, add the results to the description of this PR
5. once this PR is merged and a new version of vaultclient is released, update vault2 with the new version of vaultclient

**Review by XinLi0207 — APPROVED** [review]

File: lib/IAMClient.d.ts [hunk 1]
```diff
@@ -92,11 +93,11 @@ declare class VaultClient {
      * Update Quota of an account
      *
      * @param {string} accountName - account name
-     * @param {integer} quota - maximum quota for the account
+     * @param {Number} quota - maximum quota for the account
```
Comment (ghost) [inline lib/IAMClient.d.ts:96]: Nit:
```suggestion
     * @param {number} quota - maximum quota for the account
```

Alternatively having the types with typescript in a jsdoc is optional, the IDE will display them correctly anyway

File: lib/IAMClient.js [hunk 2]
```diff
@@ -278,7 +278,7 @@ class VaultClient {
      * Update Quota of an account
      *
      * @param {string} accountName - account name
-     * @param {integer} quota - maximum quota for the account
+     * @param {Number} quota - maximum quota for the account
```
Comment (ghost) [inline lib/IAMClient.js]:
```suggestion
     * @param {number} quota - maximum quota for the account
```

**Review by ghost — APPROVED** [review]
Quick changes but otherwise LGTM

File: lib/IAMClient.js [hunk 3]
```diff
@@ -316,19 +316,19 @@ class VaultClient {
...
-        assert(typeof name === 'string' && name !== '',
+    updateAccountAttributes(accountName, customAttributes, callback) {
+        assert(typeof accountName === 'string' && accountName !== '',
             'name parameter is required for account name');
         assert(typeof customAttributes === 'object');
         this.request('POST', '/', true, callback, {
             Action: 'UpdateAccountAttributes',
             Version: '2010-05-08',
-            name,
+            accountName,
```
Comment (anurag4DSB) [inline lib/IAMClient.js]: We need this as `name` due to OSIS.
If you want to add a type check, an alternate might be:
```suggestion
            name: accountName,
```

In Vault2 we do not support this API yet so tests will not run.

**Review by anurag4DSB — CHANGES_REQUESTED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Pull request merged**